indico/queries/workflow.py

```diff
@@ -280,3 +281,59 @@ def __init__(
         self, workflow_id: int, files: List[str] = None, urls: List[str] = None
     ):
         super().__init__(workflow_id, files=files, urls=urls, submission=True)
+
+
+class _AddDataToWorkflow(GraphQLRequest):
+    query = """
+        mutation addDataToWorkflow($workflowId: Int!) {
+            addDataToWorkflow(workflowId: $workflowId){
+                subsetId
+                workflow{
+                    id
+                    name
+                    status
+                }
+            }
+        }
+    """
+
+    def __init__(self, workflow_id: int):
+        super().__init__(
+            self.query,
+            variables={"workflowId": workflow_id},
+        )
+
+    def process_response(self, response) -> Workflow:
+        return Workflow(
+            **super().process_response(response)["addDataToWorkflow"]["workflow"]
+        )
+
+
+class AddDataToWorkflow(RequestChain):
+    """
+    Mutation to update data in a workflow, pressumably
+    after new data is added to the dataset.
+
+    Args:
+        workflow (int|Workflow): Workflow or workflow id to update
+
+    Options:
+        wait (bool, default=False): Block while polling for status of update
+
+    Returns:
+        Workflow: Updated Workflow object
+    """
+
+    def __init__(self, workflow: int, wait: bool = False):
+        self.workflow_id = workflow.id if isinstance(workflow, Workflow) else workflow
```

Comment:
I'm kind of sad this is how we've treated workflow_id. 
I think ideally for me , we'd have self.workflow and a self.workflow_id instead of having workflow_id potentially be a whole ass workflow

Reply:
yea, tbh i didn't think it made much sense but just followed what was there. might do what you say anyways, though. in general, i think that the queries need to stick to either id or the object, just cause having both is pretty confusing from a usability perspective (totally unclear why I would be able to use a workflow in one query but not another), etc

Reply:
fundamentally i think it comes down to whether we want customers to think in terms of engineering or "Indico as a product"

from an engineering pov, makes way more sense to just use the id. from a ux one, id say the whole object (even if the guts just use the id), just because the actions are being done on Workflows, not workflow ids